UM/ColorImage.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2022 Ultimaker B.V.
+# Uranium is released under the terms of the LGPLv3 or higher.
+
+from PyQt6.QtCore import QUrl, pyqtProperty, pyqtSignal
+from PyQt6.QtGui import QColor
+from PyQt6.QtQuick import QQuickPaintedItem
+from PyQt6.QtSvg import QSvgRenderer
+from UM.Application import Application
+
+# This is meant as a potential replacement for RecolorImage.
+# As we had issues with upgrading to qt6 on windows with the shader, this was developed as an alternative
+class ColorImage(QQuickPaintedItem):
+    def __init__(self, parent = None):
+        super().__init__(parent)
+        self._source = QUrl()
+        self._color = QColor()
+        self._svg_data = b""
+        self._renderer = None
+
+    sourceChanged = pyqtSignal()
+    colorChanged = pyqtSignal()
+
+    def _updateSVG(self):
+        if not self._source or self._source.toLocalFile() == "":
+            return
+        with open(self._source.toLocalFile(), "rb") as f:
+            self._svg_data = f.read()
+        self._svg_data = self._svg_data.replace(b"<svg ", b"<svg fill=\"%s\" " % self._color.name().encode("utf-8"))
+        self._renderer = QSvgRenderer(self._svg_data)
+
+    def setSource(self, source):
```

Comment:
```suggestion
    def setSource(self, source: Optional[Any[Path, str, QUrl]]) -> None:
```
if source is indeed this type

Reply:
Unfortunately it is not this type :'(